src/api/c/random.cpp

```diff
@@ -118,29 +121,23 @@ af_err af_create_random_engine(af_random_engine *engineHandle, af_random_engine_
     try {
         AF_CHECK(af_init());
         validateRandomType(rtype);
+
         RandomEngine e;
```

Comment:
I know you didn't add this but the way the handles are created seems inefficient. Could you look into improving it?

Reply:
Sure.